From: zhang jiao <zhangjiao2@cmss.chinamobile.com>
Subject: tools/accounting/procacct: Fix minor errors
Date: Tue, 3 Dec 2024 10:05:50 +0800

The logfile option was documented but not working.  Add it and optimized
the while loop.

Link: https://lkml.kernel.org/r/20241203020550.3145-1-zhangjiao2@cmss.chinamobile.com
Signed-off-by: zhang jiao <zhangjiao2@cmss.chinamobile.com>
Reviewed-by: Dr. Thomas Orgis <thomas.orgis@uni-hamburg.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/accounting/procacct.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/tools/accounting/procacct.c~tools-accounting-procacct-fix-minor-errors
+++ a/tools/accounting/procacct.c
@@ -274,12 +274,11 @@ int main(int argc, char *argv[])
 	int maskset = 0;
 	char *logfile = NULL;
 	int cfd = 0;
-	int forking = 0;
 
 	struct msgtemplate msg;
 
-	while (!forking) {
-		c = getopt(argc, argv, "m:vr:");
+	while (1) {
+		c = getopt(argc, argv, "m:vr:w:");
 		if (c < 0)
 			break;
 
_
